django/db/models/sql/compiler.py

```diff
@@ -307,7 +308,9 @@ def get_order_by(self):
                 order_by.append((field, False))
                 continue
             if field == '?':  # random
-                order_by.append((OrderBy(Random()), False))
+                # Not reference, but should be treated as one to avoid the
+                # GROUP BY clause.
+                order_by.append((OrderBy(Random()), None))
```

Comment:
I don't think that is the right solution, I would rather add `get_group_by_cols()` to the `Random()` function:
```python
class Random(NumericOutputFieldMixin, Func):

    ...

    def get_group_by_cols(self, alias=None):
        return []
```

Reply:
Cool! I didn't think of `.get_group_by_cols()` initially!